### PR TITLE
Stage master

### PR DESCRIPTION
**Pull request by tarun-gujjula**

### CODE REVIEW

File: VerloopSDK/VLConfig.swift [hunk 1]
```diff
@@ -94,25 +184,23 @@ public typealias LiveChatUrlClickListener = (_ url : String?)  -> Void
...
-            customFields.append(CustomField(key: key, value: value, scope: "user"))
-        case .ROOM:
-            customFields.append(CustomField(key: key, value: value, scope: "room"))
+        if !key.hasEmptyValue(),!value.hasEmptyValue() {
+            updatedConfigParams.append(.customFields)
+            switch scope {
+                case .USER:
+                    customFields.append(CustomField(key: key, value: value, scope: "user"))
+                case .ROOM:
+                    customFields.append(CustomField(key: key, value: value, scope: "room"))
```
Comment (raghav39) [inline VerloopSDK/VLConfig.swift:194]: If the scope is not provided, it is room scope

File: VerloopSDK/VLExtensions.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+extension String {
+    
+    func hasEmptyValue() -> Bool {
+        return self.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty || self == ""
+    }
+    
+    static func getUserIdEvaluationJS(_ userId:String) -> String {
+//        if let _unwrapped = optionArgument {
```
Comment (raghav39) [inline VerloopSDK/VLExtensions.swift:20]: remove commented

File: VerloopSDK/VLWebViewManager.swift [hunk 3]
```diff
@@ -157,3 +237,243 @@ class VLWebViewManager: NSObject, WKScriptMessageHandler, WKUIDelegate, WKNaviga
...
+//                            self?._eventDelegate?.didEventOccurOnLiveChat(.onChatMinimized)
+                        }
+                    }
+                default: break
+            }
+        }
+        roomReadyConfigurations = []
+    }
+    
+    private func processNetworkChangeConfigurations() {
```
Comment (raghav39) [inline VerloopSDK/VLWebViewManager.swift:264]: check this

File: VerloopSDK/VerloopSDK.swift [hunk 4]
```diff
@@ -137,89 +211,89 @@ import Foundation
...
-        public let type: String
-        public let payload: String
+        public let title: String?
+        public let type: String?
+        public let payload: String?
     }
     
     private struct OnURLClick: Decodable {
-        public let url: String
+        public var title:String?
```
Comment (raghav39) [inline VerloopSDK/VerloopSDK.swift:295]: on url click, we support just the url as string

File: VerloopSDK/VLConfig.swift (hunk 1 above)
Comment (tarun-gujjula) [reply VerloopSDK/VLConfig.swift:194]: Its an enum with two possible values. I'd added a default on the switch case.

File: VerloopSDK/VerloopSDK.swift (hunk 4 above)
Comment (tarun-gujjula) [reply VerloopSDK/VerloopSDK.swift:295]: From the looks of it, minor refactoring to be done here. I believe the payload contains the url which is being passed to the callback on the front. Will debug this case subsequently.

File: VerloopSDK/VLWebViewManager.swift (hunk 3 above)
Comment (tarun-gujjula) [reply VerloopSDK/VLWebViewManager.swift:264]: A subset of callbacks are triggered on network change. Hence we are using this method. Commenting here to keep track.

**Pull request merged**